{"name": "Тест оформления заказа залогиненным пользователя", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //p[text()='Флюоресцентная булка R2-D3']; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16", "trace": "self = <tests.test_main_function.TestMainFunction object at 0x104f65bb0>, auth_user = <selenium.webdriver.firefox.webdriver.WebDriver (session=\"e762227a-2156-48a2-ad2f-b90c057a0d5c\")>\n\n    @allure.title(\"Тест оформления заказа залогиненным пользователя\")\n    @allure.description(\"Залогиненный пользователь может оформить заказ\")\n    def test_make_order_by_auth_account(self, auth_user):\n        driver = auth_user\n        main_page = MainPage(driver)\n>       main_page.add_ingredient()\n\ntests/test_main_function.py:68: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/main_page.py:33: in add_ingredient\n    self.drag_and_drop_element(MainPageLocators.BUN_INGREDIENT, MainPageLocators.BASKET)\npages/base_page.py:77: in drag_and_drop_element\n    drag = self.driver.find_element(*locator_1)\n.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:748: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py:354: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x104f8fd00>, response = {'status': 404, 'value': '{\"value\":{\"error\":\"no such element\",\"message\":\"Unable to locate element: //p[text()=\\'Флюоре.../content/shared/webdriver/Errors.sys.mjs:511:5\\\\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //p[text()='Флюоресцентная булка R2-D3']; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nE       Stacktrace:\nE       RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nE       WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nE       NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\nE       dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n.venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py:229: NoSuchElementException"}, "description": "Залогиненный пользователь может оформить заказ", "steps": [{"name": "Добавление ингредиента в заказ", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //p[text()='Флюоресцентная булка R2-D3']; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n", "trace": "  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/pages/main_page.py\", line 33, in add_ingredient\n    self.drag_and_drop_element(MainPageLocators.BUN_INGREDIENT, MainPageLocators.BASKET)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/pages/base_page.py\", line 77, in drag_and_drop_element\n    drag = self.driver.find_element(*locator_1)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 748, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 354, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "steps": [{"name": "Перетаскивание элемента", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //p[text()='Флюоресцентная булка R2-D3']; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception\nStacktrace:\nRemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8\nWebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5\nNoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5\ndom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16\n\n", "trace": "  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/pages/base_page.py\", line 77, in drag_and_drop_element\n    drag = self.driver.find_element(*locator_1)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 748, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py\", line 354, in execute\n    self.error_handler.check_response(response)\n  File \"/Users/p.sinegubov/Downloads/Личные данные/git/Диплом_Павел_Синегубов_8/Diplom_3/.venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "parameters": [{"name": "locator_1", "value": "('xpath', \"//p[text()='Флюоресцентная булка R2-D3']\")"}, {"name": "locator_2", "value": "('class name', 'BurgerConstructor_basket__list__l9dp_')"}], "start": 1719513716729, "stop": 1719513716731}], "start": 1719513716729, "stop": 1719513716731}], "parameters": [{"name": "driver", "value": "'firefox'"}], "start": 1719513716729, "stop": 1719513716731, "uuid": "897474f6-3ff4-4857-9731-b7c0b47b6b2b", "historyId": "82412c04f14678c150813bd4d44f3eb7", "testCaseId": "9cf0da10b45c87ffc77066bda28a83e2", "fullName": "tests.test_main_function.TestMainFunction#test_make_order_by_auth_account", "labels": [{"name": "feature", "value": "Проверка основного функционала"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_main_function"}, {"name": "subSuite", "value": "TestMainFunction"}, {"name": "host", "value": "NB0727.local"}, {"name": "thread", "value": "44469-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_main_function"}]}